src/lib/JsResolver/JsResolverSecrets.ts

```diff
@@ -0,0 +1,111 @@
+import { Signer } from "@ethersproject/abstract-signer";
+import axios, { Axios } from "axios";
+import { getAuthToken, SiweOverride } from "./authToken";
+import { OPS_USER_API } from "../../constants";
+import { Secrets } from "../../types";
+import { errorMessage } from "../../utils";
+
+export class JsResolverSecrets {
+  private readonly _signer: Signer;
+  private readonly _userApi: Axios;
+
+  constructor(signer: Signer) {
+    this._signer = signer;
+    this._userApi = axios.create({
+      baseURL: OPS_USER_API,
+    });
+  }
+
+  public async get(key: string, override?: SiweOverride): Promise<string> {
+    try {
+      const address = await this._signer.getAddress();
+      const authToken = await getAuthToken(this._signer, override);
+
+      const res = await this._userApi.get(`/users/${address}/secrets/${key}`, {
+        headers: { Authorization: `Bearer ${authToken}` },
+      });
+
+      const secret = res.data[key];
+      if (secret === undefined)
+        throw new Error(`Secret not found for key: ${key}`);
```

Comment:
let's just return `undefined` if it's not found
it's a bit more flexible for end users than throwing an exception

Reply:
[dabf76a](https://github.com/gelatodigital/ops-sdk/pull/25/commits/dabf76ae4e4ae22d9e6128fab9436fe25c9c6904)